verilog/tools/ls/autoexpand_test.cc

```diff
@@ -0,0 +1,258 @@
+  TestTextEdits(GenerateAutoExpandTextEdits,
+                R"(
+module t1(/*AUTOARG*/);
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/);
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)",
+                R"(
+module t1(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)");
+}
+
+TEST(Autoexpand, ExpandReplace) {
+  TestTextEdits(GenerateAutoExpandTextEdits,
+                R"(
+module t(/*AUTOARG*/
+  //Inputs
+  clk, rst
+// some comment
+);
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule)",
+                R"(
+module t(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule)");
+}
+
+TEST(Autoexpand, SkipPredeclared) {
+  TestTextEdits(GenerateAutoExpandTextEdits,
+                R"(
+module t(i,
+         o1, /*AUTOARG*/
+//Inputs
+clk, rst
+);
+  input logic clk;
+  input logic rst;
+  input logic i;
+  output logic o1;
+  output logic o2;
+endmodule)",
+                R"(
+module t(i,
+         o1, /*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o2
+  );
+  input logic clk;
+  input logic rst;
+  input logic i;
+  output logic o1;
+  output logic o2;
+endmodule)");
+}
+
+TEST(Autoexpand, CodeActionExpandAll) {
+  TestTextEdits(
+      [](BufferTracker* tracker) {
+        return AutoExpandCodeActionToTextEdits(tracker, {},
+                                               "Expand all AUTOs in file");
+      },
+      R"(
+module t1(/*AUTOARG*/);
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/);
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)",
+      R"(
+module t1(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)");
+}
+
+TEST(Autoexpand, CodeActionExpandRange) {
+  TestTextEdits(
+      [](BufferTracker* tracker) {
+        return AutoExpandCodeActionToTextEdits(
+            tracker, {.start = {.line = 0}, .end = {.line = 1}},
+            "Expand this AUTO");
+      },
+      R"(
+module t1(/*AUTOARG*/);
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/);
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)",
+      R"(
+module t1(/*AUTOARG*/
+  // Inputs
+  clk,
+  rst,
+  // Outputs
+  o
+  );
+  input logic clk;
+  input logic rst;
+  output logic o;
+endmodule
+module t2(/*AUTOARG*/);
+  input logic clk;
+  input rst;
+  output logic o;
+endmodule
+)");
+}
+
```

Comment:
Can you add another test that adds an `/*AUTOARG*/` in some situation that is not in the argument place ? So nothing should be expanded there and also not upset the expander.